### ISSUE TITLE
ELIFECYCLE error

### ISSUE THREAD

**Issue by bklynraised** [edwarddistel/mta-subway-countdown-clock#8]
Hi!
I'm afraid this is something very basic or easy, but I can't get the script to run past this:

```
Error: You must enter a valid API key from the MTA.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! mta-countdown-clock@1.0.0 start: `node ./dist/bundle.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the mta-countdown-clock@1.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/mccormackterence/.npm/_logs/2021-02-05T19_08_47_439Z-debug.log

```

I have definitely edited the .env file to include my API, even tried getting a new one. 
Any hints?

Sorry if this is 101 stuff!

**Comment by bklynraised** [comment]
Well, I've started fresh and discovered that the errors start before then.
From the Console: 

> ERROR in [at-loader] ./src/mta.ts:13:9 
>     TS2554: Expected 0 arguments, but got 1.
> npm ERR! code ELIFECYCLE
> npm ERR! errno 2
> npm ERR! mta-countdown-clock@1.0.0 build: `webpack --config ./webpack.config.js`
> npm ERR! Exit status 2
> npm ERR! 
> npm ERR! Failed at the mta-countdown-clock@1.0.0 build script.
> npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
> 
> npm ERR! A complete log of this run can be found in:
> npm ERR!     /Users/mccormackterence/.npm/_logs/2021-02-05T19_52_17_525Z-debug.log


 
The Log:
> 0 info it worked if it ends with ok
> 1 verbose cli [ '/usr/local/bin/node', '/usr/local/bin/npm', 'run', 'build' ]
> 2 info using npm@6.14.10
> 3 info using node@v14.15.4
> 4 verbose run-script [ 'prebuild', 'build', 'postbuild' ]
> 5 info lifecycle mta-countdown-clock@1.0.0~prebuild: mta-countdown-clock@1.0.0
> 6 info lifecycle mta-countdown-clock@1.0.0~build: mta-countdown-clock@1.0.0
> 7 verbose lifecycle mta-countdown-clock@1.0.0~build: unsafe-perm in lifecycle true
> 8 verbose lifecycle mta-countdown-clock@1.0.0~build: PATH: /usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/Users/mccormackterence/Downloads/mta-subway-countdown-clock-master/node_modules/.bin:/Users/mccormackterence/.yarn/bin:/Users/mccormackterence/.config/yarn/global/node_modules/.bin:/opt/local/bin:/opt/local/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin
> 9 verbose lifecycle mta-countdown-clock@1.0.0~build: CWD: /Users/mccormackterence/Downloads/mta-subway-countdown-clock-master
> 10 silly lifecycle mta-countdown-clock@1.0.0~build: Args: [ '-c', 'webpack --config ./webpack.config.js' ]
> 11 silly lifecycle mta-countdown-clock@1.0.0~build: Returned: code: 2  signal: null
> 12 info lifecycle mta-countdown-clock@1.0.0~build: Failed to exec build script
> 13 verbose stack Error: mta-countdown-clock@1.0.0 build: `webpack --config ./webpack.config.js`
> 13 verbose stack Exit status 2
> 13 verbose stack     at EventEmitter.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
> 13 verbose stack     at EventEmitter.emit (events.js:315:20)
> 13 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
> 13 verbose stack     at ChildProcess.emit (events.js:315:20)
> 13 verbose stack     at maybeClose (internal/child_process.js:1048:16)
> 13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5)
> 14 verbose pkgid mta-countdown-clock@1.0.0
> 15 verbose cwd /Users/mccormackterence/Downloads/mta-subway-countdown-clock-master
> 16 verbose Darwin 20.3.0
> 17 verbose argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "build"
> 18 verbose node v14.15.4
> 19 verbose npm  v6.14.10
> 20 error code ELIFECYCLE
> 21 error errno 2
> 22 error mta-countdown-clock@1.0.0 build: `webpack --config ./webpack.config.js`
> 22 error Exit status 2
> 23 error Failed at the mta-countdown-clock@1.0.0 build script.
> 23 error This is probably not a problem with npm. There is likely additional logging output above.
> 24 verbose exit [ 2, true ]
>

**Comment by edwarddistel** [comment]
@bklynraised The app should always build and run, even if there is an invalid API key. Sounds like something went wrong run the installation of the dependencies. Did you run `npm run i` to install them?

That being said, it'd been a while since I looked at this app and it turns out the app, unrelated to your issue, was not actually working -- the MTA completely revised their APIs. So while I was at it I updated the app to reflect their new APIs.

You'll now also have to select an endpoint, see the updated README docs.

But I just tested the app and it now should be able to run fine.
```
npm run i
npm run build
npm run start
```

Then open http://localhost:4000
